bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/AccessibleToolBar.java

```diff
@@ -0,0 +1,77 @@
+package org.eclipse.ui.texteditor;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.TraverseEvent;
+import org.eclipse.swt.events.TraverseListener;
+import org.eclipse.swt.graphics.Color;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.ToolBar;
+import org.eclipse.swt.widgets.ToolItem;
+
+import org.eclipse.jface.layout.GridLayoutFactory;
+
+/**
+ * This class wraps the ToolBar to make it possible to use tabulator-keys to
+ * navigate between the buttons of a ToolBar. For this, we simulate a singular
+ * ToolBar by putting each ToolItem into it's own ToolBar and composing them
+ * into a Composite. Since the "Enter" keypress could not previously trigger
+ * activation behavior, we listen for it manually and send according events if
+ * necessary.
+ *
+ * @since 3.18
+ */
+class AccessibleToolBar extends Composite {
+
+	private List<ToolBar> toolBars = new ArrayList<>();
+
+	public AccessibleToolBar(Composite parent) {
+		super(parent, SWT.NONE);
+		GridLayoutFactory.fillDefaults().numColumns(0).spacing(0, 0).margins(0, 0).applyTo(this);
+	}
+
+	/**
+	 * Creates a ToolItem handled by this ToolBar and returns it. Will add a
+	 * KeyListener which will catch all "Enter"-Keypresses.
+	 *
+	 * @param styleBits the StyleBits to apply to the created ToolItem
+	 * @return a newly created ToolItem
+	 */
+	public ToolItem createToolItem(int styleBits) {
+		ToolBar parent = new ToolBar(this, SWT.FLAT | SWT.HORIZONTAL);
+		ToolItem result = new ToolItem(parent, styleBits);
```

Comment:
I'm not a big fan of calling variables "result" or the likes.

```suggestion
		ToolItem toolItem = new ToolItem(parent, styleBits);
```

Reply:
I don't believe this improves the code quality, but I don't mind performing this change.

Reply:
changed in https://github.com/eclipse-platform/eclipse.platform.ui/commit/0c0d22bc8f0ea13eb6e1124c89747f7a4f02e266